dbms/src/AggregateFunctions/AggregateFunctionGroupConcat.h

```diff
@@ -116,32 +116,36 @@ class AggregateFunctionGroupConcat final
 
     DataTypePtr getReturnType() const override { return result_is_nullable ? makeNullable(ret_type) : ret_type; }
 
-    /// reject nulls before add() of nested agg
-    void add(AggregateDataPtr __restrict place, const IColumn ** columns, size_t row_num, Arena * arena) const override
+    template <bool is_add>
+    void addOrDecrease(AggregateDataPtr __restrict place, const IColumn ** columns, size_t row_num, Arena * arena) const
     {
         if constexpr (only_one_column)
         {
             if (is_nullable[0])
             {
-                const ColumnNullable * column = static_cast<const ColumnNullable *>(columns[0]);
+                const auto * column = static_cast<const ColumnNullable *>(columns[0]);
                 if (!column->isNullAt(row_num))
                 {
                     this->setFlag(place);
                     const IColumn * nested_column = &column->getNestedColumn();
-                    this->nested_function->add(this->nestedPlace(place), &nested_column, row_num, arena);
+
+                    if constexpr (is_add)
+                        this->nested_function->add(this->nestedPlace(place), &nested_column, row_num, arena);
+                    else
+                        this->nested_function->decrease(this->nestedPlace(place), &nested_column, row_num, arena);
```

Comment:
Ditto

Reply:
> Ditto

Because `addOrDecrease<>()` is not an interface in `IAggregateFunction`.